### ISSUE TITLE
Incremental build fails in nightly

### ISSUE THREAD

**Issue by stepancheg** [rust-lang/rust#37792]
```
% git clone https://github.com/rust-lang-nursery/bitflags
% cd bitflags
% RUSTFLAGS="-Zincremental=$HOME/tmp/grpc-rust" RUST_BACKTRACE=1 cargo build --verbose
   Compiling bitflags v0.7.0 (file:///Users/yozh/devel/left/bitflags)
     Running `rustc src/lib.rs --crate-name bitflags --crate-type lib -g -C metadata=6898b5682b95a1aa --out-dir /Users/yozh/devel/left/bitflags/target/debug/deps --emit=dep-info,link -L dependency=/Users/yozh/devel/left/bitflags/target/debug/deps -Zincremental=/Users/yozh/tmp/grpc-rust`
error: internal compiler error: unexpected panic

note: the compiler unexpectedly panicked. this is a bug.

note: we would appreciate a bug report: https://github.com/rust-lang/rust/blob/master/CONTRIBUTING.md#bug-reports

note: run with `RUST_BACKTRACE=1` for a backtrace

thread 'rustc' panicked at 'no entry found for key', ../src/libcore/option.rs:705
stack backtrace:
   1:        0x10afca4ea - std::sys::imp::backtrace::tracing::imp::write::h944c02ac40aee2d7
   2:        0x10afd777f - std::panicking::default_hook::{{closure}}::h6875a2976258b020
   3:        0x10afd732d - std::panicking::default_hook::h88ffbc5922643264
   4:        0x10afd7c46 - std::panicking::rust_panic_with_hook::ha5aed1dfc0e220e3
   5:        0x10afd7ae4 - std::panicking::begin_panic::hc066339e2fdc17d1
   6:        0x10afd7a02 - std::panicking::begin_panic_fmt::h5912b2d2df332044
   7:        0x10afd7967 - rust_begin_unwind
   8:        0x10b019aa0 - core::panicking::panic_fmt::h561c5ee168a3d2cb
   9:        0x10b019b0d - core::option::expect_failed::h6be288f6c3caa41a
  10:        0x107893018 - <rustc_incremental::calculate_svh::IncrementalHashesMap as core::ops::Index<&'a rustc::dep_graph::dep_node::DepNode<rustc::hir::def_id::DefId>>>::index::hfb9f68bd7c3a82d2
  11:        0x1078a22b6 - rustc_incremental::persist::hash::HashContext::hash::h9d2f063fe38bcd58
  12:        0x1078af0a9 - rustc_incremental::persist::save::save_dep_graph::h8ad44fda4512c5f8
  13:        0x1068d4250 - rustc_driver::driver::phase_4_translate_to_llvm::h48ed91c172294403
  14:        0x1068a46bc - rustc_driver::driver::compile_input::{{closure}}::hf13172fc4e5a2f4d
  15:        0x1068c211b - rustc_driver::driver::phase_3_run_analysis_passes::{{closure}}::h97a3a12d948df547
  16:        0x1068bb37d - rustc_driver::driver::phase_3_run_analysis_passes::hb0ad9de18d423e67
  17:        0x1068a2bc5 - rustc_driver::driver::compile_input::h8e119234b60571d5
  18:        0x1068e8229 - rustc_driver::run_compiler::hbdfc4f84e2e0f4b9
  19:        0x106807988 - std::panicking::try::do_call::hf679f17bf3b43b0b
  20:        0x10afda1fa - __rust_maybe_catch_panic
  21:        0x10682baaf - <F as alloc::boxed::FnBox<A>>::call_box::h506fb5d7b8891cd4
  22:        0x10afd68d4 - std::sys::imp::thread::Thread::new::thread_start::h8084b1107992ae5b
  23:     0x7fff991a999c - _pthread_body
  24:     0x7fff991a9919 - _pthread_start

error: Could not compile `bitflags`.

Caused by:
  process didn't exit successfully: `rustc src/lib.rs --crate-name bitflags --crate-type lib -g -C metadata=6898b5682b95a1aa --out-dir /Users/yozh/devel/left/bitflags/target/debug/deps --emit=dep-info,link -L dependency=/Users/yozh/devel/left/bitflags/target/debug/deps -Zincremental=/Users/yozh/tmp/grpc-rust` (exit code: 101)
zsh: exit 101   RUSTFLAGS="-Zincremental=$HOME/tmp/grpc-rust" RUST_BACKTRACE=1 cargo build
% rustc --version
rustc 1.15.0-nightly (0ed951993 2016-11-14)
% uname
Darwin
```

**Comment by AndyBarron** [comment]
I'm also getting this error 😢

**Comment by TimNN** [comment]
I think this is a duplicate of #37756 which should be fixed by #37787.

**Comment by michaelwoerister** [comment]
Thanks for the bug report! It is indeed the same error as #37756. Closing as duplicate.
